tests/functional/lifecycle/LifecycleTask.js

```diff
@@ -0,0 +1,511 @@
+};
+
+/*
+    NOTE:
+    When using CURRENT, running the lifecycle tasks will internally use
+    `Date.now` to compare dates. This means that CURRENT date usage will most
+    likely be in the past.
+    To avoid flakiness, I will be setting the day of CURRENT back 2 day to
+    avoid any flakiness.
+    When comparing rules, if I set a rule that says "Days: 1", then any objects
+    using CURRENT as LastModified should pass. To avoid flakiness, set a rule
+    that says "Days: 3" to have any objects using CURRENT as LastModified to
+    not pass.
+    Also, when using rules that require Days/NoncurrentDays/DaysAfterInitiation,
+    XML rule restrictions don't allow "0" day rules to be applied. So when
+    testing, always use ENV `TEST_SWITCH` and LifecycleTask will expire based
+    on 1 Day rules. Applying 2+ Days should mean the object(s) should not
+    expire.
+*/
+// current date
+const CURRENT = new Date();
+CURRENT.setDate(CURRENT.getDate() - 2);
+// 5 days prior to currentDate
+const PAST = new Date(CURRENT);
+PAST.setDate(PAST.getDate() - 5);
+// 5 days after currentDate
+const FUTURE = new Date(CURRENT);
+FUTURE.setDate(FUTURE.getDate() + 5);
+
+const OWNER = 'testOwner';
+
+class S3Helper {
+    constructor(client) {
+        this.s3 = client;
+        this.bucket = undefined;
+
+        this._scenario = [
+            // 0. create objects: non-versioned, no pagination
+            {
+                keyNames: ['object-1', 'object-2', 'object-3'],
+                tags: ['key1=value1', 'key1=value1&key2=value2', 'key2=value2'],
+
+            },
+            // 1. create objects: non-versioned, pagination, prefix
+            {
+                keyNames: [
+                    'test/obj-1', 'obj-2', 'test/obj-3',
+                    'atest/obj-4', 'testx/obj-5', 'test/obj-6',
+                    'obj-7', 'atest/obj-8', 'an-obj-9',
+                ],
+                tags: [
+                    'key1=value1', 'key1=value1', 'key1=value1&key2=value2',
+                    'key2=value2', 'key2=value2', 'key1=value1&key3=value3',
+                    'key3=value3', 'key3=value3', 'key4=value4',
+                ],
+            },
+        ];
+    }
+
+    getBucket() {
+        return this.bucket;
+    }
+
+    setAndCreateBucket(name, cb) {
+        this.bucket = name;
+        this.s3.createBucket({
+            Bucket: name,
+        }, err => {
+            assert.ifError(err);
+            cb();
+        });
+    }
+
+    createObjects(scenarioNumber, cb) {
+        async.forEachOf(this._scenario[scenarioNumber].keyNames,
+        (key, i, done) => {
+            this.s3.putObject({
+                Body: '',
+                Bucket: this.bucket,
+                Key: key,
+                Tagging: this._scenario[scenarioNumber].tags[i],
+            }, done);
+        }, err => {
+            assert.ifError(err);
+            return cb();
+        });
+    }
+
+    emptyAndDeleteBucket(cb) {
```

Comment:
We have this implemented as `empty()` function in S3 repo at tests/functional/aws-node-sdk/lib/utility/bucket-util.js, ideally we would share one implementation across projects, for now it's fine.